azkaban-common/src/main/java/azkaban/version/JdbcVersionSetLoader.java

```diff
@@ -0,0 +1,237 @@
+   *
+   * @param dbOperator Interface to perform db related operations.
+   */
+  @Inject
+  public JdbcVersionSetLoader(DatabaseOperator dbOperator) throws IOException {
+    this.dbOperator = dbOperator;
+    for (VersionSet versionSet : fetchAllVersionSets()) {
+      this.md5ToVersionSet.put(versionSet.getVersionSetMd5Hex(), versionSet);
+      this.idToVersionSet.put(versionSet.getVersionSetId(), versionSet);
+    }
+  }
+
+  /**
+   * This method inserts the versionSetJsonString and versionSetMd5Hex to the Table version_set and
+   * get the versionSetId autogenerated post insertion.
+   */
+  @Override
+  public synchronized int insertAndGetVersionSetId(String versionSetMd5Hex,
+      String versionSetJsonString) throws IOException {
+    final SQLTransaction<Integer> insertVersionSet = transOperator -> {
+      final VersionSetHandler versionSetHandler = new VersionSetHandler();
+      // Check if the versionSet already exists for versionSetMd5Hex
+      List<VersionSet> versionSets = transOperator
+          .query(SELECT_VSET_FROM_MD5, versionSetHandler, versionSetMd5Hex);
+      if (versionSets.size() > 1) {
+        throw new SQLException(
+            "Expected only one VersionSet for versionSetMd5Hex " + versionSetMd5Hex);
+      }
+      // Insert if it doesn't
+      if (versionSets.isEmpty()) {
+        transOperator.update(INSERT_VSET, versionSetMd5Hex, versionSetJsonString);
+        return (int) transOperator.getLastInsertId();
+      }
+      return versionSets.get(0).getVersionSetId();
+    };
+    try {
+      Integer versionSetId = this.dbOperator.transaction(insertVersionSet);
+      VersionSet versionSet = new VersionSet(versionSetJsonString, versionSetMd5Hex, versionSetId);
+      this.md5ToVersionSet.put(versionSetMd5Hex, versionSet);
+      this.idToVersionSet.put(versionSetId, versionSet);
+      return versionSetId;
+    } catch (SQLException e) {
+      logger.error("Exception occurred while inserting version set and getting version id", e);
+      throw new IOException("Unable to insert and get versionSetId", e);
+    }
+  }
+
+  /**
+   * Deletes the versionSet row from the table version_set corresponding to versionSetMd5Hex.
+   */
+  @Override
+  public synchronized boolean deleteVersionSet(String versionSetMd5Hex) throws IOException {
+    final SQLTransaction<Integer> insertVersionSet = transOperator -> transOperator
+        .update(DELETE_VSET, versionSetMd5Hex);
+    try {
+      Integer rowsDeleted = this.dbOperator.transaction(insertVersionSet);
+      if (this.md5ToVersionSet.containsKey(versionSetMd5Hex)) {
+        VersionSet removedVersionSet = this.md5ToVersionSet.remove(versionSetMd5Hex);
+        this.idToVersionSet.remove(removedVersionSet.getVersionSetId());
+      }
+      return rowsDeleted != 0;
+    } catch (SQLException e) {
+      logger.error("Failed to execute: " + DELETE_VSET, e);
+      throw new IOException("Exception occurred while removing version set for " + versionSetMd5Hex,
```

Comment:
Same as above.

Reply:
Since VersionSetLoader interface has the definition to throw IOException, it is kept along the same lines.